CXP-1150: Added documentaion for new update app scopes endpoint (#575)
akeneo/pim-api-docs: content/apps/update-scopes.md

# Update access scopes

::: warning
**This feature is only available on SaaS platforms.**
:::

It's possible to update the access scopes of an already connected app.

> The access scope update process requires app users to consent to new scopes changes. 

Reviewer: ```suggestion
The access scope update process requires app users to consent to new scopes addition. 
```

If your app needs to change the initial scope list, it must initiate an authorization request with the entire list of access scopes your app needs. The authorization request follows the usual OAuth 2.0 protocol through which the app will be able to retrieve a new Access Token that reflects new requested scopes.

Through Akeneo REST API, connected apps can notify Akeneo PIM users who can manage your app that the authorization scopes your app requires have changed. 

To do so, use the following `POST` method, with parameters below:

```
https://my-pim.cloud.akeneo.com/connect/apps/v1/scopes/update?scope=[REQUESTED_SCOPES]
```
Requirements:

> - `scopes`: query parameter to specify app new scope list as a space-separated string

Reviewer: provide an example ?

Reviewer 2: I don't think that's necessary.

- `Authorization: Bearer [AccessToken]`: header to authenticate your app
- `Content-Type: application/json`: header for the response format

Response: `"Ok"`

> The notification will warn them and entice them to open your app so that you can initiate an authorization request with the entire list of required scopes.

Reviewer: maybe display a screenshot of the notification ?

Reviewer 3: yes now i have a screenshot ;)

# Update access scopes

::: warning
**This feature is only available on SaaS platforms.**
:::

It's possible to update the access scopes of an already connected app.
The access scope update process requires app users to consent to new scopes addition. 

If your app needs to change the access scopes, it must initiate a new authorization request with all access scopes needed, even the access scopes that are already granted. This new authorization request follows the usual OAuth 2.0 protocol and will end with a new Access Token that reflects updated scopes.

::: warning
**Some users are not allowed to grant new access scopes in the Akeneo PIM**.  
You should not force users to go through the Authorization process if you haven't received the new access scopes, you could end up in an infinite loop. Consequently, your App must be able to function as usual without the new access scopes.
  

> More information on our Help Center: [Who can manage and open Apps?](https://help.akeneo.com/pim/serenity/articles/manage-your-apps.html#who-can-manage-and-open-apps)

Reviewer 3: Our English teacher told me to write on our Help Center.

:::

## Notify a PIM your app requires an authorization update

Through Akeneo REST API, connected apps can notify Akeneo PIM users who can manage your app that the authorization scopes your app requires have changed. 

To do so, use the following `POST` method, with parameters below:

```

> https://my-pim.cloud.akeneo.com/connect/apps/v1/scopes/update?scope=[REQUESTED_SCOPES]

Reviewer: ```suggestion
https://my-pim.cloud.akeneo.com/connect/apps/v1/scopes/update?scopes=[REQUESTED_SCOPES]
```

# Update access scopes

::: warning
**This feature is only available on SaaS platforms.**
:::

> It's possible to update the access scopes of an already connected app.
> The access scope update process requires app users to consent to new scopes addition. 
> 
> If your app needs to change the initial scope list, it must initiate an authorization request with the entire list of access scopes your app needs. The authorization request follows the usual OAuth 2.0 protocol through which the app will be able to retrieve a new Access Token that reflects new requested scopes.

Reviewer 2: ```suggestion
If your app needs to change the access scopes, it must initiate a new authorization request with all access scopes needed, even the access scopes that are already granted. This new authorization request follows the usual OAuth 2.0 protocol and will end with a new Access Token that reflects updated scopes.
```

::: warning
**Not all users of Akeneo PIM can manage scopes changes.**  

> Some users might only have the rights to use your app but not manage it on the Akeneo PIM side. For this reason, it is up to your app to be able to function as per usual until it retrieves a new Access Token.  

Reviewer 2: ```suggestion
**Some users are not allowed to grant new access scopes in the Akeneo PIM**. You should not force users to go through the Authorization process if you haven't received the new access scopes, you could end up in an infinite loop. Consequently, your App must be able to function as usual without the new access scopes.
```

  
More information on our Help Center: [Who can manage and open Apps?](https://help.akeneo.com/pim/serenity/articles/manage-your-apps.html#who-can-manage-and-open-apps)
:::

## Notify a PIM your app requires an authorization update

Through Akeneo REST API, connected apps can notify Akeneo PIM users who can manage your app that the authorization scopes your app requires have changed. 

To do so, use the following `POST` method, with parameters below:

```
https://my-pim.cloud.akeneo.com/connect/apps/v1/scopes/update?scope=[REQUESTED_SCOPES]
```
Requirements:
- `scopes`: query parameter to specify app new scope list as a space-separated string
- `Authorization: Bearer [AccessToken]`: header to authenticate your app
- `Content-Type: application/json`: header for the response format

> Response: `"Ok"`

Reviewer 2: ```suggestion
Response: `"Ok"`
```

IMHO, it's useless to document it.

Author: We don't have an explicit reference for this endpoint either

Reviewer 2: That's ok, it's already the case for other Apps endpoints and it would be useless outside the scopes context.
